var/spack/repos/builtin/packages/opencv/package.py

```diff
@@ -1,364 +1,406 @@
+    version('3.4.5',    sha256='0c57d9dd6d30cbffe68a09b03f4bebe773ee44dc8ff5cd6eaeb7f4d5ef3b428e')
+    version('3.4.4',    sha256='a35b00a71d77b484f73ec485c65fe56c7a6fa48acd5ce55c197aef2e13c78746')
     version('3.4.3',    sha256='4eef85759d5450b183459ff216b4c0fa43e87a4f6aa92c8af649f89336f002ec')
     version('3.4.1',    sha256='f1b87684d75496a1054405ae3ee0b6573acaf3dad39eaf4f1d66fdd7e03dc852')
     version('3.4.0',    sha256='678cc3d2d1b3464b512b084a8cca1fad7de207c7abdf2caa1fed636c13e916da')
     version('3.3.1',    sha256='5dca3bb0d661af311e25a72b04a7e4c22c47c1aa86eb73e70063cd378a2aa6ee')
     version('3.3.0',    sha256='8bb312b9d9fd17336dc1f8b3ac82f021ca50e2034afc866098866176d985adc6')
-    version('3.2.0',    sha256='9541efbf68f298f45914b4e837490647f4d5e472b4c0c04414a787d116a702b2')
-    version('3.1.0',    sha256='f3b160b9213dd17aa15ddd45f6fb06017fe205359dbd1f7219aad59c98899f15')
-    version('2.4.13.2', sha256='4b00c110e6c54943cbbb7cf0d35c5bc148133ab2095ee4aaa0ac0a4f67c58080')
-    version('2.4.13.1', sha256='0d5ce5e0973e3a745f927d1ee097aaf909aae59f787be6d27a03d639e2d96bd7')
-    version('2.4.13',   sha256='94ebcca61c30034d5fb16feab8ec12c8a868f5162d20a9f0396f0f5f6d8bbbff')
-    version('2.4.12.3', sha256='a4cbcd2d470860b0cf1f8faf504619c18a8ac38fd414c5a88ed3e94c963aa750')
-    version('2.4.12.2', sha256='150a165eb14a5ea74fb94dcc16ac7d668a6ff20a4449df2570734a2abaab9c0e')
-    version('2.4.12.1', sha256='c1564771f79304a2597ae4f74f44032021e3a46657e4a117060c08f5ed05ad83')
-
-    # Standard variants
+    version('3.2.0',    sha256='9541efbf68f298f45914b4e837490647f4d5e472b4c0c04414a787d116a702b2', deprecated=True)
+    version('3.1.0',    sha256='f3b160b9213dd17aa15ddd45f6fb06017fe205359dbd1f7219aad59c98899f15', deprecated=True)
+    version('2.4.13.2', sha256='4b00c110e6c54943cbbb7cf0d35c5bc148133ab2095ee4aaa0ac0a4f67c58080', deprecated=True)
+    version('2.4.13.1', sha256='0d5ce5e0973e3a745f927d1ee097aaf909aae59f787be6d27a03d639e2d96bd7', deprecated=True)
+    version('2.4.13',   sha256='94ebcca61c30034d5fb16feab8ec12c8a868f5162d20a9f0396f0f5f6d8bbbff', deprecated=True)
+    version('2.4.12.3', sha256='a4cbcd2d470860b0cf1f8faf504619c18a8ac38fd414c5a88ed3e94c963aa750', deprecated=True)
+    version('2.4.12.2', sha256='150a165eb14a5ea74fb94dcc16ac7d668a6ff20a4449df2570734a2abaab9c0e', deprecated=True)
+    version('2.4.12.1', sha256='c1564771f79304a2597ae4f74f44032021e3a46657e4a117060c08f5ed05ad83', deprecated=True)
+
+    # OpenCV modules (variants)
+    # Defined in `modules/*/CMakeLists.txt` using
+    # `ocv_add_module(...)` and `ocv_define_module(...)`
+    modules = [
+        'apps', 'calib3d', 'core', 'dnn', 'features2d', 'flann', 'gapi',
+        'highgui', 'imgcodecs', 'imgproc', 'java', 'java_bindings_generator',
+        'js', 'js_bindings_generator', 'ml', 'objc', 'objc_bindings_generator',
+        'objdetect', 'photo', 'python2', 'python3',
```

Comment:
I meant before to update from upstream what is essential. Were all the bumps needed?

Anyways, there is now no `+python` variant, but a `+python3` variant. You'll have to change the ML packages from `opencv+python` to `opencv+python3` (IMO splitting the variant like that was a bad decision…)

Reply:
About the bumps, I started with pulling `mxnet`, and after that more things stopped working so I pulled more. The dependencies can be summaried like that:
```
py-atlinter
└── mxnet
    ├── opencv
    │   └── onnx
    └── py-graphvz
```
Maybe there was a way of minimally adjusting `mxnet` by hand without pulling everything from upstream.

Reply:
Re `+python`, nice find. I grepped to find where this could happend and I'm seeing this:
```
var/spack/repos/builtin/packages/candle-benchmarks/package.py:28:    depends_on('opencv@3.2.0: +core +highgui +imgproc +jpeg +png +tiff +zlib +python -dnn ~eigen ~gtk')
var/spack/repos/builtin/packages/py-cinema-lib/package.py:40:    depends_on('opencv@3.4:+python', when='+opencv', type=('build', 'run'))
var/spack/repos/builtin/packages/py-imgaug/package.py:28:    depends_on('opencv +python', type=('run'))
var/spack/repos/builtin/packages/py-mmcv/package.py:23:    depends_on('opencv+python', type=('build', 'run'))
...
```
I guess these would all have problems with `opencv` lacking the `+python` variant?

Reply:
Yes. I'd only change the ones you use, to avoid merge conflicts down the line.

And I'd rather see the bumps in the dependencies than having to manually merge conflicts in `mxnet` later :-/